usaspending_api/awards/serializers_v2/serializers.py

```diff
@@ -15,3 +19,425 @@ class RecipientAwardSpendingSerializer(serializers.Serializer):
+        model = LegalEntityOfficers
+        fields = '__all__'
+        default_fields = [
+            "officer_1_name",
+            "officer_1_amount",
+            "officer_2_name",
+            "officer_2_amount",
+            "officer_3_name",
+            "officer_3_amount",
+            "officer_4_name",
+            "officer_4_amount",
+            "officer_5_name",
+            "officer_5_amount",
+        ]
+
+
+class ToptierAgencySerializerV2(LimitableSerializerV2):
+
+    class Meta:
+        model = ToptierAgency
+        fields = '__all__'
+        default_fields = [
+            "name",
+            "abbreviation"
+        ]
+
+
+class SubtierAgencySerializerV2(LimitableSerializerV2):
+
+    class Meta:
+        model = SubtierAgency
+        fields = '__all__'
+        default_fields = [
+
+            "name",
+            "abbreviation"
+        ]
+
+
+class OfficeAgencySerializerV2(LimitableSerializerV2):
+
+    class Meta:
+        model = OfficeAgency
+        fields = '__all__'
+        default_fields = [
+            "aac_code",
+            "name"
+        ]
+
+
+class LocationSerializerV2(LimitableSerializerV2):
+
+    class Meta:
+        model = Location
+        fields = '__all__'
+        default_fields = [
+            "address_line1",
+            "address_line2",
+            "address_line3",
+            "foreign_province",
+            "city_name",
+            "county_name",
+            "state_code",
+            "zip5",
+            "zip4",
+            "foreign_postal_code",
+            "country_name",
+            "location_country_code",
+            "congressional_code"
+        ]
+
+
+class AgencySerializerV2(LimitableSerializerV2):
+    office_agency_name = serializers.SerializerMethodField('office_agency_name_func')
+
+    def office_agency_name_func(self, agency):
+        if agency.office_agency:
+            return agency.office_agency.name
+        else:
+            return None
+
+    class Meta:
+        model = Agency
+        fields = '__all__'
+        default_fields = [
+            "toptier_agency",
+            "subtier_agency",
+            "office_agency_name",
+        ]
+        nested_serializers = {
+            "toptier_agency": {
+                "class": ToptierAgencySerializerV2,
+                "kwargs": {"read_only": True}
+            },
+            "subtier_agency": {
+                "class": SubtierAgencySerializerV2,
+                "kwargs": {"read_only": True}
+            },
+            "office_agency": {
+                "class": OfficeAgencySerializerV2,
+                "kwargs": {"read_only": True}
+            },
+        }
+
+
+class LegalEntitySerializerV2(LimitableSerializerV2):
+    recipient_parent_name = serializers.SerializerMethodField('recipient_parent_name_func')
+
+    def recipient_parent_name_func(self, entity):
+        try:
+            parent_recipient = DUNS.objects.get(awardee_or_recipient_uniqu=entity.parent_recipient_unique_id)
```

Comment:
DUNS is a good source of recipient data, unfortunately it doesn't contain 100% of the recipient data we have in the DB. `recipient_lookup` is a better table in terms of data completeness. However, there hasn't been clear direction concerning when it is OK to use these "normalized" data vs the submitted data. Waiting for a response from POs